### ISSUE TITLE
Query result for embedded document is different

### ISSUE THREAD

**Issue by marchpig** [bwaldvogel/mongo-java-server#35]
Hi,

I found that the query result for embedded document is different with the real DB.
The following test is failed as the `1st` document is not found by the query.
```java
@Test
public void test() {
    mongoTemplate.insert(new A()); // 1st
    mongoTemplate.insert(new A(new B())); // 2nd
    mongoTemplate.insert(new A(new B(new C()))); // 3rd

    List<A> list = mongoTemplate.find(Query.query(Criteria.where("b.c").is(null)), A.class);

    Assert.assertEquals(2, list.size()); // Failed here [Expected:2  / Actual:1]
    list.forEach(a -> Assert.assertTrue(a.b == null || a.b.c == null));
}
```
```java
public class A {
    public B b;

    public A() {}
    public A(B b) {
        this.b = b;
    }
}
```
```java
public class B {
    public C c;

    public B() {}
    public B(C c) {
        this.c = c;
    }
}
```
```java
public class C {
}
```

Thanks!

**Comment by bwaldvogel** [comment]
Thanks for reporting. The bug should be fixed with f8d1a93. Can you retest?

**Comment by marchpig** [comment]
@bwaldvogel,
Thanks a lot for the quick fix!
But f8d1a93 couldn't fix this bug, so I created a PR(#37).

**Comment by bwaldvogel** [comment]
Thanks!

**Comment by bwaldvogel** [comment]
Fix is released in version `1.10.0`
